### ISSUE TITLE
Expecting closingReason to notify.

### ISSUE THREAD

**Issue by robertavram** [PolymerElements/paper-dialog-behavior#77]
When using an implementation of the paper-dialog-behaviour (such as paper-dialog element) the api notes that the ```closingReason``` object is set, however since this property does not notify it becomes difficult to understand it's use case.

My expectation was to have a property or an event that notifies or triggers when the user confirms or dismisses a modal.

One potential current method could be to use ```iron-overlay-closed``` and then read the ```closingReason``` property too see if a modal was dismissed or confirmed, however this becomes cumbersome and confusing (especially when using multiple dialogs or nested modals) since one either has to stop the event propagation or read properties of the ```currentTarget``` in order to determine which dialog triggered the event.

A potential solution could be for the ```closingReason``` object to be set with the setter set('object.property', 'closingReason') as well as enabling notification on the object.

therefore as a user of the element I can: 
```
<paper-dialog closing-reason="{{dialogOneReason}}"> ---buttons go here --- </paper-dialog>
...
...
//set an observer
observers: [ 'dialogActionTaken(dialogOneReason)']
```

**Comment by valdrinkoshi** [comment]
Hi @robertavram, could you provide an example where the current events are not enough?
In the current implementation, `dialog-dismiss` and `dialog-confirm` will always close the dialog and `iron-overlay-closed` will provide the `closingReason`; this event will also tell you which dialog triggered it (`event.target`).
Consider that `iron-overlay-behavior` also sets `closingReason.canceled`, and you won't get notified about that.

**Comment by robertavram** [comment]
Hi @valdrinkoshi,
I ran into an issue when i was using dialogs nested inside different elements. All of a sudden, after I closed one dialog all my functions upstream were called, and like you said, parallel events need to check their target and then apply the logic correspondingly. The functionality is present, and I ended up doing just that in the end, as well as stoping the event propagation to the parent elements.

Diving into the code a little deeper I realized that having the canceledReason notify is not a good idea at all since there can cases in which the dialog is closed and the canceledReason does not change, therefore no notification.

Sorry to waste your time :) I've had a long week... :)
